src/main.rs

```diff
@@ -211,7 +213,6 @@ fn main() {
         .add_startup_system(set_audio_channels_volume)
         .add_enter_system(GameState::InGame, play_level_music)
         .add_exit_system(GameState::InGame, stop_level_music)
-        .add_exit_system(GameState::InGame, clean_in_game_data)
```

Comment:
Is this the change which basically prevents cleanup from happening everytime `GameState::InGame` exits? This is the fix and the rest is mostly just cleanup correct?

Reply:
Yep, this is the fix.

The rest of the code just makes the logic re-usable.

Reply:
Cool let's get it in